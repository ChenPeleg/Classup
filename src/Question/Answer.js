/* one question */
import React from "react";

const answer = props => {
  return <p className="answer">{props.content}</p>;
};
export default answer;
