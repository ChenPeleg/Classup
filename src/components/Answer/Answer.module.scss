.answer {
  position: relative;
  background-color: antiquewhite;
  color: rgb(24, 20, 20);
  width: 70%;
  left: 15%;

  background: rgb(2, 0, 36);
  background: rgb(2, 0, 36);
  background: rosybrown;
  border-radius: 9px;
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.5);
  margin: 15px;
  padding: 5px;
}
.choosenAnswer {
  background: rgb(223, 200, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 20, 36, 0.9);
}

.answer:not(.choosenAnswer):hover {
  cursor: pointer;
  background: rgb(223, 175, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.9);
}
.answer:active {
  cursor: pointer;
  background: rgb(223, 200, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 20, 36, 0.9);
}
