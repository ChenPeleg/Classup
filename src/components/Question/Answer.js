/* one question */
import React from "react";

const answer = props => {
  return (
    <p
      className="answer"
      onClick={event => props.answerHandler(event, +props.num + 1)}
    >
      {props.content}
    </p>
  );
};
export default answer;
