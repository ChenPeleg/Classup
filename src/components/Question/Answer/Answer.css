.answer {
  position: relative;
  background-color: antiquewhite;
  color: rgb(24, 20, 20);
  width: 70%;
  left: 15%;
  height: 60%;
  background: rgb(2, 0, 36);
  background: rgb(2, 0, 36);
  background: rosybrown;
  border-radius: 9px;
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.5);
  margin: 15px;
  padding: 5px;
}
.choosenAnswer {
  background: rgb(223, 200, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 20, 36, 0.9);
}

.answer:not(.choosenAnswer):hover {
  cursor: pointer;
  background: rgb(223, 175, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 0, 36, 0.9);
}
.answer:active {
  cursor: pointer;
  background: rgb(223, 200, 175);
  box-shadow: 2px 2px 4px 2px rgba(2, 20, 36, 0.9);
}

.check-mark-wrapper {
  position: absolute;
  width: 50px;
  height: 50px;
  top: -10px;
  left: -10px;
}
.check-mark {
  position: relative;
  width: 100%;
  height: 100%;
}
