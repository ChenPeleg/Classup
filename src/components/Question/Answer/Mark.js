import styles from "./Mark.module.scss";
import React from "react";
import Vmark from "../../../assets/images/check-mark.svg";
import Xmark from "../../../assets/images/x-mark.svg";

const mark = (props) => {
  return (
    <div className={styles.checkMarkWrapper}>
      <img
        src={props.marked_Answer === "RIGHT" ? Vmark : Xmark}
        className={styles.checkMark}
        alt="MyBurger"
      />
    </div>
  );
};
export default mark;
