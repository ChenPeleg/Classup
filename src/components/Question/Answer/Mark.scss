.check-mark-wrapper {
  position: absolute;
  width: 50px;
  height: 50px;
  top: -10px;
  left: -10px;
}
.check-mark {
  position: relative;
  width: 100%;
  height: 100%;
  opacity: 0;
  animation: check-animate 0.3s forwards;
}
@keyframes check-animate {
  0% {
    opacity: 0;
  }
  50% {
    opacity: 0.5;
  }
  100% {
    opacity: 1;
  }
}
