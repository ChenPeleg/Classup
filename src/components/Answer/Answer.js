/* one question */
import React from "react";
import styles from "./Answer.module.scss";
import Mark from "../Mark/Mark";

const answer = (props) => {
  return (
    <div
      className={`${styles.answer} ${
        props.isChosen ? styles.choosenAnswer : ""
      }`}
      onClick={(event) => {
        props.chooseAnswerHandler(event, +props.num);
      }}
    >
      {props.content}
      {props.isChosen && props.marked_Answer ? (
        <Mark marked_Answer={props.marked_Answer} />
      ) : null}
    </div>
  );
};
export default answer;
