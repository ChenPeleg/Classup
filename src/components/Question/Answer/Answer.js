/* one question */
import React from "react";
import "./Answer.css";
import Mark from "./Mark";

const answer = props => {
  return (
    <div
      className={`answer ${props.isChosen ? "choosenAnswer" : ""}`}
      onClick={event => props.answerHandler(event, +props.num)}
    >
      {props.content}
      {props.isChosen && props.marked_Answer ? (
        <Mark marked_Answer={props.marked_Answer} />
      ) : null}
    </div>
  );
};
export default answer;
