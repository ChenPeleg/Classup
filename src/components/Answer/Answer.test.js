import React from "react";
import { shallow } from "enzyme";
import Answer from "./Answer";
import Mark from "../Mark/Mark";

const mockChoose = jest.fn();
const ansNum = 5;
const answer = shallow(<Answer isChosen={true} marked_Answer={"RIGHT"} num={ansNum} chooseAnswerHandler={mockChoose} />);

describe("<Answer>", () => {

  it("renders correctly", () => {
    expect(answer).toMatchSnapshot();
  });

  describe("<Answer> choosen with marks", () => {

    it("should render a check mark", () => {
      expect(answer.contains(<Mark />));
    });
  });
  describe("<Answer> is clicked on", () => {

    beforeEach(() => {
      answer.simulate("click");
    });
    it("calls a function with the ChooseAnswer callback", () => {
      expect(mockChoose).toHaveBeenCalledWith(undefined, ansNum);
    });
  });
});
