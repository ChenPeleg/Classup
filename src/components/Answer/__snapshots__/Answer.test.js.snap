// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Answer> renders correctly 1`] = `
<div
  className="answer choosenAnswer"
  onClick={[Function]}
>
  <mark
    marked_Answer="RIGHT"
  />
</div>
`;
